NumberPlayScanner/src/NumberPlayScanner.java

```diff
@@ -0,0 +1,154 @@
+import java.util.ArrayList;
+import java.util.List;
+
+import org.opencv.core.Core;
+import org.opencv.core.Mat;
+import org.opencv.core.MatOfKeyPoint;
+import org.opencv.core.MatOfPoint;
+import org.opencv.core.Point;
+import org.opencv.core.Rect;
+import org.opencv.core.Scalar;
+import org.opencv.core.Size;
+//import org.opencv.features2d.FeatureDetector;
+//import org.opencv.features2d.Features2d;
+import org.opencv.imgcodecs.Imgcodecs;
+import org.opencv.imgproc.Imgproc;
+
+public class NumberPlayScanner{
+  Mat image;
+  Mat grayscaleImage;
+  Mat binImage;
+  Mat boundaryDeletedImage;
+  
+  List<MatOfPoint> contours;
+  Mat hierarchy;
+  List<Point> sudokuPoints;
+  
+  MatOfKeyPoint detectedPoints;
+  
+  public NumberPlayScanner(Mat image){
+    this.image = image;
+    this.grayscaleImage = new Mat();
+    this.binImage = new Mat();
+    this.boundaryDeletedImage = new Mat();
+    
+    this.contours = new ArrayList<MatOfPoint>();
+    this.hierarchy = new Mat();
+    
+    this.detectedPoints = new MatOfKeyPoint();
+  }
+  
+  public void run(){
+    imageBinarization();
+    
+    contourDetection();
+    
+    boundaryDeletion();
+    
+    divideCells();
+  }
+  
+  private void imageBinarization(){
+    Imgproc.cvtColor(this.image, grayscaleImage, Imgproc.COLOR_RGB2GRAY);
+    
+    //obtain an image easier to do image processing by gaussian blur
+    Imgproc.GaussianBlur(this.grayscaleImage, this.grayscaleImage, new Size(5, 5), 1, 1);
+    
+    //Adaptive thresholding is used. Need to find most appropriate number for the last two ints of inputs
+    Imgproc.adaptiveThreshold(this.grayscaleImage, this.binImage, 255,
+                              Imgproc.ADAPTIVE_THRESH_GAUSSIAN_C, Imgproc.THRESH_BINARY, 7, 10);
+    Core.bitwise_not(this.binImage, this.binImage);
+    Imgproc.dilate(this.binImage, this.binImage, new Mat());
+    
+    //debug process for binary image
+    writeImage(this.grayscaleImage, "output/grayscale.png");
+    writeImage(this.binImage, "output/binary.png");
+  }
+  
+  private void contourDetection(){
+    Mat contourDetectedImage = this.image.clone();
+    //copy and paste code, working fine currently. Should look into it to find better suited logic
+    Imgproc.findContours(this.binImage, this.contours, this.hierarchy,
+                         Imgproc.RETR_EXTERNAL, Imgproc.CHAIN_APPROX_TC89_L1);
+    
+    //debug process for contour detection
+    Imgproc.drawContours(contourDetectedImage, this.contours, -1, new Scalar(0, 0, 255), 3);
+    writeImage(contourDetectedImage, "output/detectedContours.png");
+    
+    System.out.println(String.format("%d\n", this.contours.size()));
+    for(int i = 0; i < this.contours.size(); i++){
+      System.out.println("" + this.contours.get(i).dump());
```

Comment:
I actually didn't recognize the situation, but it seems this bit of code is somehow doing nothing at all.
I do remember when many contours were detected it worked, so I rewrite it as your comment suggests and leave this code here.